packages/components/src/unit-control/index.js

```diff
@@ -107,6 +84,42 @@ function UnitControl(
 		setUnit( next );
 	};
 
+	const mayUpdateUnit = ( event ) => {
+		if ( ! isNaN( event.target.value ) ) {
+			refParsedValue.current = null;
+			return;
+		}
+		const [ parsedValue, parsedUnit ] = getValidParsedUnit(
+			event.target.value,
+			units,
+			value,
+			unit
+		);
+
+		refParsedValue.current = parsedValue;
+
+		if ( isPressEnterToChange && parsedUnit !== unit ) {
+			const data = CSS_UNITS.find(
```

Comment:
I believe this needs to change to `units` (from props) rather than `CSS_UNITS`.
The reason is because the collection of units can be customized :).
In the case of Gutenberg, it would be via the `theme.json` file.

Reply:
Glad you caught that! It's updated in 09a4127495dc66eea57b9e3a196e489c4d4fb35b